### DISCUSSION TITLE
"error in loading plugin" - How do I debug this?

### DISCUSSION THREAD

**Discussion by theoparis** [outfoxxed/hy3#44] [Q&A] [answered]
```
hyprctl plugin load /usr/local/lib/libhy3.so
error in loading plugin
```

There is no other error message besides this.

I built hyprland and hy3 from the latest commit on a Debian sid system.

**Comment by outfoxxed** [comment]
Check the hyprland log, Ctrl+F `libhy3.so`.

This usually means there is a version mismatch between hy3 and hyprland. Ensure you are using master for hyprland-git, or the correct tag for a release build.

**Reply by theoparis** [reply]
I just ran git checkout hl0.32.0 on hy3 and v0.32.0 on Hyprland and rebuilt, and I have the same issue still 🤔 
```
Hyprland, built from branch HEAD at commit 751d2851cc270c3322ffe2eb83c156e4298a0c0e dirty (props: bump ver to 0.32.0).
Tag: v0.32.0

flags: (if any)
debug
```

The logs say invalid dispatcher, not much else.

**Reply by outfoxxed** [reply]
That is the correct commit at least. I'm going to need the logs to do anything further.

Run `tail /tmp/hypr/$HYPRLAND_INSTANCE_SIGNATURE/hyprland.log -f` and then attempt to load the plugin in another terminal. Copy the output from the first one here so I can check it.

**Reply by outfoxxed** [reply]
did you solve this?

**Reply by theoparis** [reply, marked as answer]
> $HYPRLAND_INSTANCE_SIGNATURE/hyprland.log -f



> That is the correct commit at least. I'm going to need the logs to do anything further.
> 
> Run `tail /tmp/hypr/$HYPRLAND_INSTANCE_SIGNATURE/hyprland.log -f` and then attempt to load the plugin in another terminal. Copy the output from the first one here so I can check it.

In this log file it says it is already loaded when I try to load it. However, If i try to set up the mouse or keyboard bindings to use hy3:movewindow it doesn't let me do anything - movewindow works fine.

Edit: Sorry I totally skipped over an important part of the configuration - I forgot to set general:layout to hy3 :joy:
